lib/Dialect/TritonGPU/Transforms/OptimizeDotOperands.cpp

```diff
@@ -15,6 +17,96 @@ namespace gpu {
 
 namespace {
 
+// Helpers
+
+// Returns whether we can hoist DotOp Encoding through `op`.
+// Roughly, whether op is elementwise and thus threads don't need
+// to exchange elements. But some ops are not current supported even though
+// they meet that criterion.
+bool canHoistDotOpEncV2(Operation* op, DotOperandEncodingAttr& dotOpEnc) {
+  // Only consider custom conversions or arith ops.
+  // TODO(jlebar): Is this too restrictive?
+  if (!isa<FpToFpOp, BitcastOp>(op) && !isPureUnaryInlineAsm(op) &&
+      op->getDialect()->getTypeID() != TypeID::get<arith::ArithDialect>())
```

Comment:
I'm aware this code was only moved but `!isa<arith::ArithDialect>(op->getDialect())` would be better.